### PR TITLE
Use `jnr-posix-api` plugin

### PR DESCRIPTION
**Pull request by basil**
This PR switches this plugin to depend on the newly-released [`jnr-posix-api`](https://plugins.jenkins.io/jnr-posix-api/) plugin rather than getting JNR from core. Once this is merged and released, and once enough users have upgraded to this release, we can consider removing the JNR dependency from core.

CC @daniel-beck @Wadeck

### CODE REVIEW

**Comment by daniel-beck** [comment]
If the plugin isn't detached, wouldn't it make more sense to use a different package? How confident are we about classloading correctness, potentially involving dynamically loaded plugins, when classes are in core and a plugin?

**Comment by basil** [comment]
> How confident are we about classloading correctness, potentially involving dynamically loaded plugins, when classes are in core and a plugin?

I haven't done an end-to-end test but I'm pretty confident. I added

https://github.com/jenkinsci/jnr-posix-api-plugin/blob/fbb90b6ad669feef44cd5bcf548b8080d1e5c9d7/pom.xml#L70

to `jnr-posix-api-plugin` and did some testing in the script console to make sure the class loading worked as expected with both older and newer cores. Once the baseline is updated to a core that includes jenkinsci/jenkins#5979 I plan to remove everything except `org.objectweb.asm.` from that line.

In terms of `pam-auth` specifically, it doesn't have a `<maskClasses>` entry, so it will find the class from core right now, and once jenkinsci/jenkins#5979 is merged and released it'll fail to find it in core and then look in the plugin's class loader and find it there through `jnr-posix-api-plugin`. So there should be no problem.

**Comment by basil** [comment]
Any next steps for moving forward with this?

**Comment by basil** [comment]
@daniel-beck @Wadeck ?

**Comment by jglick** [comment]
> it doesn't have a `<maskClasses>` entry, so it will find the class from core right now

I guess there is no chance of using `RealJenkinsRule` due to the problem of mocking out PAM?

**Review by jglick — APPROVED** [review]
Sounds OK. (I apparently have GH write access, but not Artifactory access.)

**Comment by basil** [comment]
@daniel-beck @Wadeck Gentle ping.

**Comment by daniel-beck** [comment]
I see this updates a bundled JNA, probably due to the core dependency update and presence in the core BOM. Does this library need to be bundled here, or can we just use whatever is in core?

**Comment by basil** [comment]
> Does this library need to be bundled here, or can we just use whatever is in core?

JNA does not need to be bundled here, nor is it used at runtime in production because core provides JNA already. In any case this is a pre-existing issue not related to this PR.

**Pull request merged**
